swarm/src/lib.rs

```diff
@@ -1508,38 +1510,44 @@ impl error::Error for DialError {
     }
 }
 
-/// Dummy implementation of [`NetworkBehaviour`] that doesn't do anything.
-#[derive(Clone)]
-pub struct DummyBehaviour {
-    keep_alive: KeepAlive,
-}
+/// Implementation of [`NetworkBehaviour`] that doesn't do anything other than keep all connections alive.
+pub struct KeepAliveBehaviour;
+
+impl NetworkBehaviour for KeepAliveBehaviour {
+    type ConnectionHandler = KeepAliveConnectionHandler;
+    type OutEvent = void::Void;
 
-impl DummyBehaviour {
-    pub fn with_keep_alive(keep_alive: KeepAlive) -> Self {
```

Comment:
Just for the record, I do **not** think we need to role this out in two steps, (1) deprecating it and (2) removing it.